components/feature/prompts/src/main/java/mozilla/components/feature/prompts/ext/BrowserStore.kt

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.prompts.ext
+
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.selector.findTabOrCustomTabOrSelectedTab
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.prompt.PromptRequest
+import mozilla.components.feature.prompts.dialog.PromptDialogFragment
+import java.lang.ref.WeakReference
+
+internal fun BrowserStore.consumePromptFrom(
+    sessionId: String?,
+    activePrompt: WeakReference<PromptDialogFragment>? = null,
+    consume: (PromptRequest) -> Unit
+) {
+    if (sessionId == null) {
+        state.selectedTab
+    } else {
+        state.findTabOrCustomTabOrSelectedTab(sessionId)
+    }?.let { tab ->
+        activePrompt?.clear()
+        tab.content.promptRequest?.let {
+            consume(it)
+            dispatch(ContentAction.ConsumePromptRequestAction(tab.id))
```

Comment:
I think we are migrating away from consuming prompts by sessions instead we are using UIDs because https://github.com/mozilla-mobile/android-components/issues/8989

You can see some recently work made by @Mugurell on
https://github.com/mozilla-mobile/android-components/commit/95f4cb010cf3b543b8d5877e3c58ad1c1d98dc12#diff-e172d93510635621502da071e27011e020186c4c3922b4efb1efdef3e8f959bdR470